### PR TITLE
Add additional Android requirements to README

### PR DESCRIPTION
**Pull request by bidetofevil**
Modify README to reference an additional requirement for Android projects that support Android versions that require desugaring. Details are on this [Slack thread](https://cloud-native.slack.com/archives/C014L2KCTE3/p1713392437479949).

I can't find a style guide that tells me how this modification should be formatted, so please let me know if this doesn't work and I'll fix it up!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,18 @@ To use these artifacts, you must also depend on `io.opentelemetry:opentelemetry-
...
+> 
+> If you are using this on Android and your project's minSdk is lower than 26, you must enable
+> [desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring).
+> See [#73](https://github.com/open-telemetry/opentelemetry-android/issues/73) for more information.
+> 
+> If your project's minSdk is lower than 24, in order to run instrumentation tests or run the app
+> built on debug, you must use AGP 8.3.0+ and set the `android.useFullClasspathForDexingTransform`
+> property in `gradle.properties` to `true` to ensure desugaring runs properly. For the full
+> context for this workaround, please see
+> [this issue](https://issuetracker.google.com/issues/230454566#comment18).
```
Comment (jack-berg) [inline README.md:67]: Screenshot 2024-05-06 at 10 20 04 AM

This renders weird. Can you remove the `>` quoting and change `> Android Requirements` to `## Android Requirements`?

File: README.md [hunk 2]
```diff
@@ -54,6 +54,18 @@ To use these artifacts, you must also depend on `io.opentelemetry:opentelemetry-
 See [opentelemetry-java releases](https://github.com/open-telemetry/opentelemetry-java#releases) for
 more information.
 
+> Android Requirements
+> 
+> If you are using this on Android and your project's minSdk is lower than 26, you must enable
```
Comment (jack-berg) [inline README.md:59]: Can we just link to the core repo section which talks about desugaring? https://github.com/open-telemetry/opentelemetry-java/blob/main/VERSIONING.md#language-version-compatibility

The "minSdk is lower than 26" requirement is the type of thing what will become out of date and I'm not really sure where 26 comes from.

Comment (bidetofevil) [reply README.md:59]: 26 is Android API version code. It's pretty impenetrable unless you're deep in that ecosystem.

I'll make a quick reference to desugaring here and point to the `opentelementry-android` repo for more details so we can isolate the details in one place at least.

I'll put up a revision in the next couple days.

**Comment by bidetofevil** [comment]
Updated. I linked to the PR instead of the README for details in case the README changes. I can change that if y'all want.

**Review by jack-berg — APPROVED** [review]
Thanks!

**Comment by jack-berg** [comment]
Looks like the build is failing for something unrelated to this PR, which should be fixed in #71. After that is merged, we can merge main into this branch and merge.

**Pull request merged**